graphql_compiler/schema_transformation/rename_schema.py

```diff
@@ -684,6 +725,14 @@ class RenameSchemaTypesVisitor(Visitor):
     # type named "Foo", types_involving_interfaces_with_field_renamings will contain "Foo".
     types_involving_interfaces_with_field_renamings: Set[str]
 
+    # Collects names of interfaces to keep in the schema but make unqueryable as a result of one of
+    # their descendants in the inheritance hierarchy being suppressed. If a type named "Foo"
+    # implements at least one interface, then suppressed_types_implementing_interfaces will contain
+    # "Foo".
+    suppressed_types_implementing_interfaces: Set[
+        Union[ObjectTypeDefinitionNode, InterfaceTypeDefinitionNode]
```

Comment:
Seems you generally collect type names rather than the types themselves. What is the reason for collecting types here rather than type names?

Reply:
We want to find the interfaces that each given type implements, which we can find indirectly using the `interfaces` field of each type definition node. GraphQL schemas do have a `mapping` attribute that maps from definition name to definition node, but it'd involve an extra level of indirection.